src/OpenFeature/Model/ProviderEvents.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Generic;
+using OpenFeature.Constant;
+
+namespace OpenFeature.Model
+{
+    /// <summary>
+    /// The EventHandlerDelegate is an implementation of an Event Handler
+    /// </summary>
+    public delegate void EventHandlerDelegate(ProviderEventPayload eventDetails);
+
+    /// <summary>
+    /// Contains the payload of an OpenFeature Event.
+    /// </summary>
+    public class ProviderEventPayload
```

Comment:
Can we add a test that populates and asserts the contents of some of the additional fields here including `Message` and `FlagChanges`?